### DISCUSSION TITLE
Package not found in the Github Registry

### DISCUSSION THREAD

**Discussion by joekaiser** [community/community#25211] [Actions]
<p>I have a nodejs workflow and of course I am trying to install dependencies. One of those dependencies is in another repository’s package repo. I have the workflow authenticating just fine, but the problem is that it doesn’t find the package. </p>
<p>Am I mising something?  The package is under joekaiser/test and the workflow is running at joekaiser/ci-test</p>
<p>To be clear, I know the pacakge exists. The same project builds in AWS CodePipeline.</p>
<pre><code>npm ERR! 404 Not Found - GET https://npm.pkg.github.com/@joekaiser%2ftest - npm package "test" does not exist under owner "joekaiser"

name: BuildDeploy

on:
  push:
    branches:
      - release

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v1
    - name: Setup Node.js
      uses: actions/setup-node@v1
      with:
        node-version: 12
        registry-url: https://npm.pkg.github.com/
        scope: '@joekaiser'
    - name: npm install and build
      env:
        NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
        CI: true
      run: |
        npm install
        npm run build
</code></pre>

**Comment by ankri** [comment, marked as answer]
<p>At the moment you’ll need a personal access token for that.</p>
<p><span class="mention">@Phanatic</span>:</p>
<blockquote>
<p>Ahh, that’s what I figured. The GITHUB_TOKEN we generate is scoped to the repository that is running the Workflow. Unfortunately, this doesn’t allow us to install or publish packages from/to other repositories. I’ll bring this up in planning and we can figure out how to proceed here.</p>
</blockquote>
<p>Related <a href="https://github.com/actions/setup-node/issues/49" rel="nofollow noopener">issue</a></p>

**Comment by Alappin** [comment]
<p>This issue is still happening regardless whether one uses the GITHUB_TOKEN or the PERSONAL_ACCESS_TOKEN when the repository is <strong>Private</strong>.</p>
<p>Steps to reproduce.</p>
<ol>
<li>
<p>Create a Private repository “packages”.</p>
</li>
<li>
<p>Add a folder for a basic package e.g. framework</p>
</li>
<li>
<p>Publish the package to GPR either using actions or from the terminal. (configure .npmrc) e.g.</p>
</li>
</ol>
<p>package.json</p>
<pre><code>{
  "name": "@YOUR_ORG/framework",
  "version": "1.0.0",
  "description": "framework package",
  "repository": {
    "type": "git",
    "url": "git://github.com/YOUR_ORG/packages.git"
  },
  "publishConfig": {
    "registry":"https://npm.pkg.github.com"
  },
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" &amp;&amp; exit 1"
  },
  "dependencies": {
    "lodash": "^4.17.10"
  }
}
</code></pre>
<p>.npmrc</p>
<pre><code>//npm.pkg.github.com/:_authToken=PERSONAL_ACCESS_TOKEN
</code></pre>
<p>index.json</p>
<pre><code>console.log('framework');
</code></pre>
<ol start="4">
<li>
<p>Publish to GPR from with the framework folder.</p>
<p>npm publish</p>
</li>
<li>
<p>The publish will be successful but the package cannot be installed even when the token is used. e.g. a consumer with</p>
</li>
</ol>
<p>.npmrc</p>
<pre><code>//npm.pkg.github.com/:_authToken=PERSONAL_ACCESS_TOKEN
@YOUR_ORG:registry=https://npm.pkg.github.com/
</code></pre>
<p>package.json</p>
<pre><code>{
  "name": "consumer",
  "version": "1.0.0",
  "description": "consumer",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "start": "node index.js"
  },
  "dependencies": {
    "@YOUR_ORG/framework": "^1.0.0"
  }
}
</code></pre>
<p>index.json</p>
<pre><code>console.log('consumer');
</code></pre>
<ol start="6">
<li>
<p>You will see Errors like below: (YOUR_ORG is your organisation)</p>
<p>npm ERR! code E404<br>
npm ERR! 404 Not Found - GET <a href="https://npm.pkg.github.com/download/@YOUR_ORG/framework/1.0.0/SOME_HASH" class="inline-onebox-loading" rel="nofollow noopener">https://npm.pkg.github.com/download/@YOUR_ORG/framework/1.0.0/SOME_HASH</a><br>
npm ERR! 404<br>
npm ERR! 404 ‘<span class="mention">@YOUR_ORG</span>/framework@latest’ is not in the npm registry.<br>
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)<br>
npm ERR! 404<br>
npm ERR! 404 Note that you can also install from a<br>
npm ERR! 404 tarball, folder, http url, or git url.</p>
</li>
</ol>
<p>You will also notice that the link to the Assets download on the packages page is “NOT FOUND when you hover over it”</p>
<pre><code>https://notfound.pkg.github.com/
</code></pre>
<p>NOTE: this behaviour might not happen straight away in some cases, the NOT FOUND issue might start happening 30 to 45 minutes after the package has be published.</p>

**Comment by krystof-k** [comment]
<p>Your .npmrc is incorrect. It should look like this:</p>
<pre><code>//npm.pkg.github.com/:_authToken=${{ secrets.GITHUB_PERSONAL_TOKEN }}
</code></pre>
<p>Also you should do this only on the CI, because when someone checkouts the repo, it won’t work obviously. I have a step in the CI for this:</p>
<pre><code>- name: Set up NPM authentication
  run: echo "//npm.pkg.github.com/:_authToken=${{ secrets.GITHUB_PERSONAL_TOKEN }}" &gt;&gt; ~/.npmrc
</code></pre>

**Comment by wpitallo** [comment]
<p>Did you manage to get this working? I am getting the eact same error?</p>

**Comment by Alappin** [comment]
<p><a class="mention" href="/krystof-k">@krystof-k</a>  The instructions I wrote above are mearnt to be <em>sudo code</em> and just a minimum guide on what someone needs to do to reproduce the issue, there are <em>not mearnt to be a copy paste 100% accurate source code</em>. You will have to properly write the variables and so on, but this not the issue here.</p>
<p>To fully understand the problem, create a sample <strong>package that is private</strong> roughly as I outlined in the steps with your correct variable configs and you will end up with the issue I reported. It does not matter whether you do it through github actions or from your local. The problem still exists and needs to be addressed otherwise GPR is not useful if <strong>people cannot reliably host and consume PRIVATE packages</strong>.</p>

**Comment by WaqasIbrahim** [comment]
<p>Facing the same issue with a private repository, can’t install the package and the assets link to <a href="https://notfound.pkg.github.com/" rel="nofollow noopener">https://notfound.pkg.github.com/</a>. I changed the visibility to public and I was able to install the package. </p>

**Comment by Alappin** [comment]
<p><span class="mention">@wpitalloI</span> am still having the problem, I contacted github support about it and took a week before some guy got back to me and told me that he escalated the issue and have not heard anything since then so I am just waiting for someone from there to respond either on this thread or in the support ticket.</p>
<p>I just don’t get how something as critical as this issue slipped through their testing and quality assurance. At the very minimum, I would expect that if someone is providing a Package Registry, they ensure that it reliably satisfies both <em>public</em> and <em>private</em> use cases.</p>

**Comment by krystof-k** [comment]
<p>Hmm, this is weird. In our repos it works fine (using the PAT). But we’re using Yarn, may that be the reason?</p>

**Comment by mxxk** [comment]
<p>Since this issue evolved from the original question (and solution), see the new dedicated issue to here: <a href="https://github.community/t/github-package-registry-private-package-not-found/14500">https://github.community/t5/GitHub-API-Development-and/Github-package-registry-Private-package-not-found/m-p/40062/</a></p>

**Comment by r-julien** [comment]
<p>Hi, I had the same problem trying to deploy a private npm package on Github.</p>
<p>I solved it by changing the name in package.json</p>
<p>“name”: “<span class="mention">@MyOrganizationName</span>/MyRepoName”,<br>
“publishConfig”: {<br>
“registry”: “<a href="https://npm.pkg.github.com" rel="noopener nofollow ugc">https://npm.pkg.github.com</a>”<br>
},</p>
<p>Also in my npmrc:<br>
registry=https://registry.npmjs.org/<br>
<span class="mention">@MyOrganizationName</span>:registry=https://npm.pkg.github.com/<br>
<a href="//npm.pkg.github.com/:_authToken=**********" rel="noopener nofollow ugc">//npm.pkg.github.com/:_authToken=**********</a>(Token generated from github)</p>
<p>Hope that helps :slight_smile:</p>

**Reply by marcelobdiani** [reply]
What a save! thanks!

**Reply by caitlin-hamilton** [reply]
You have saved my sanity, thank you so much

**Comment by aturgarg** [comment]
<aside class="quote no-group" data-username="r-julien" data-post="11" data-topic="16251">
<div class="title">
<div class="quote-controls"></div>
 r-julien:</div>
<blockquote>
<p>npmrc:<br>
registry=https://registry.npmjs.org/<br>
<span class="mention">@MyOrganizationName</span>:registry=https://npm.pkg.github.com/<br>
<a href="//npm.pkg.github.com/:_authToken=**********" rel="noopener nofollow ugc">//npm.pkg.github.com/:_authToken=********** </a>(Token generated from github)</p>
</blockquote>
</aside>
<p>This works for me. Thanks.</p>

**Comment by 1nVitr0** [comment]
<p>For everyone getting a 404 response even though everything seems correct: <em>CHECK YOUR REPOSITORY FIELD</em>.</p>
<p>I just spent 2 hours changing every line of my GitHub action and package.json and in the end, there was an outdated value in the repository field in the package.json. In my opinion the 404 error is really really not a helpful error message here.</p>

**Comment by andylamax** [comment]
<p>Its the third quarter of 2121 and this issue still persists. SMH!! Any one of you remember how this was solved? I am having the EXACT same issue</p>

**Comment by joekaiser** [comment]
<p>As <a class="mention" href="/ankri">@ankri</a> said, Make sure you generate a Personal Access Token with the appropriate scopes.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/e/becba4be7ebafe721256b1fb06cce6c3e5d36718.png" data-download-href="/uploads/short-url/rdR1sS1ZcRK1YjyjwZMz9ukIPa0.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1521110/181089605-8668c0bc-012a-4636-8254-271a3895e139.png" alt="image" data-base62-sha1="rdR1sS1ZcRK1YjyjwZMz9ukIPa0" width="690" height="289" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/e/becba4be7ebafe721256b1fb06cce6c3e5d36718_2_690x289.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/e/becba4be7ebafe721256b1fb06cce6c3e5d36718_2_1035x433.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/e/becba4be7ebafe721256b1fb06cce6c3e5d36718.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/e/becba4be7ebafe721256b1fb06cce6c3e5d36718_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1114×468 50.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by andylamax** [comment]
<p>I checked every box while generating my PAT just to be sure. But still :slightly_frowning_face:</p>

**Comment by shellscape** [comment]
<p>This (@<a href="https://github.community/u/joekaiser">joekaiser</a>’s solution) was the solution that worked for me. Checking the <code>workflow</code> box allowed install to proceed. What’s strange is that the permission doesn’t describe the situation it needs to be checked for. Github really needs to expand that.</p>
